src/ucs/datastruct/ptr_map.inl

```diff
@@ -39,20 +46,26 @@ static inline void ucs_ptr_map_destroy(ucs_ptr_map_t *map)
 }
 
 /**
- * Put a pointer into the map
+ * Put a pointer into the map.
  *
- * @param [in]  map     Container.
- * @param [in]  ptr     Object pointer.
- * @param [in]  direct  if not 0, @a key cannot be directly converted into
- *                      @a ptr. Suggested to enable for extra @a ptr guard but
- *                      it can impact performance.
- * @param [out] key     Key to object pointer @ptr if operation completed
- *                      successfully otherwise value is undefined.
+ * @param [in]  map       Container.
+ * @param [in]  ptr       Object pointer.
+ * @param [in]  indirect  If not 0, @a key cannot be directly converted into
+ *                        @a ptr, it means @a key to @ptr association is wrapped
+ *                        by internal storage, e.g. hash table, and @ptr
+ *                        cannot be deleted more than once by @key. If this
+ *                        parameter is 0 then it is user responsibility to
+ *                        control put/get/del operations, i.e. life time of the
+ *                        association.
```

Comment:
`If nonzero, the pointer @a ptr is stored in the internal hash table, associated with a unique indirect key which is returned from this function. Otherwise, the returned value is the integer representation of the pointer @ptr`